### PR TITLE
Fix/fill logs

### PR DESCRIPTION
**Pull request by ycliuhw**
### Checklist

 - [x] Checked if it requires a [pylibjuju](https://github.com/juju/python-libjuju) change? No
 - [x] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR? Yes
 - [x] Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed? No
 - [x] Do comments answer the question of why design decisions were made? 

----

## Description of change

Rewrite `fill-unit` action/function for `fill-logs` testing charm (previously the function write a huge amount of data into stdout causing uniter memory leak);

Driveby fix: ensure function hook commands working correctly;

## QA steps

```console
# bootstrap
$ juju bootstrap aws/ap-southeast-2 k1 --debug

# deploy fill-logs
$ juju add-model t1 aws && juju deploy acceptancetests/repository/charm/fill-logs --debug

# run function
$ export JUJU_DEV_FEATURE_FLAGS=juju-v3
$ juju call fill-logs/0 fill-unit megs=100 -m k1:t1 --max-wait=2m

# check log
$ juju ssh -m k1:t1 0
$ watch ls -al /var/log/juju/unit-fill-logs*
```

## Documentation changes

No

## Bug reference

No

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -209,7 +209,7 @@ push-operator-image: operator-image
 	docker push ${OPERATOR_IMAGE_PATH}
 
 microk8s-operator-update: operator-image
-	docker save ${OPERATOR_IMAGE_PATH} | microk8s.ctr -n k8s.io image import -
+	docker save ${OPERATOR_IMAGE_PATH} | microk8s.ctr --namespac k8s.io image import -
```
Comment (wallyworld) [inline Makefile]: Why this change?
Does it have a typo?

File: acceptancetests/repository/charms/fill-logs/actions/fill-unit [hunk 2]
```diff
@@ -1,3 +1,41 @@
...
+page=$(getPage)
+pageSize=$(getStrLen ''"$page"'')
+
+while [[ "$CURRENTSIZE" -lt "$EXPECTED" ]]; do
+    action-log "writing $pageSize bytes"
+    juju-log ''"$page"''
+    let CURRENTSIZE+="$pageSize"
+done
+
+action-log "filled $CURRENTSIZE bytes log."
```
Comment (wallyworld) [inline acceptancetests/repository/charms/fill-logs/actions/fill-unit]: let's make this function-log

File: Makefile (hunk 1 above)
Comment (ycliuhw) [reply Makefile]: Recently, I got `Cannot use two forms of the same flag: n namespace` error.
Ops, a missing e..

**Comment by ycliuhw** [comment]
As discussed, we will continue to rewrite other functions of `fill-logs` charm in separate PRs.

**Comment by ycliuhw** [comment]
[Limit the size of action/function `results.Stdout`](https://trello.com/c/pWBLx2vW/980-limit-the-size-of-action-function-resultsstdout)

**Comment by ycliuhw** [comment]
$$merge$$

**Comment by ycliuhw** [comment]
$$merge$$

**Comment by ycliuhw** [comment]
$$merge$$
https://jenkins.juju.canonical.com/job/github-make-check-juju/2739/testReport/junit/github/com_juju_juju_apiserver_facades_agent_uniter/TestPackage/

**Comment by ycliuhw** [comment]
[nw-log-rotation-unit](https://trello.com/c/VXBLTmJr/121-nw-log-rotation-unit)

**Comment by ycliuhw** [comment]
https://jenkins.juju.canonical.com/job/github-make-check-juju/2741/testReport/junit/github/com_juju_juju_cmd_juju_status/TestPackage/
$$merge$$
